### PR TITLE
Closes #6175: Document how to update the tracking protection lists

### LINKED ISSUE #6175: Document how to update tracking protection lists for non-geckoView based engines

### PR DESCRIPTION
**Pull request by Amejia481**
---

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- [ ] **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- [ ] **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: docs/contribute/updating_tracking_protection_lists.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+---
+layout: page
+title: Updating the tracking protecting lists process
+permalink: /contributing/update-tracking-protection-list
+---
+
+## What is tracking protection?
+
+It is a feature that prevents trackers from collecting your personal information like your browsing habits and interests. It comes [built-in in Gecko(View)](https://mozilla.github.io/geckoview/javadoc/mozilla-central/org/mozilla/geckoview/ContentBlockingController.ContentBlockingException.html) but for no `GeckoView` engines like `SystemEngine` we have to [create our own implementation](https://github.com/vladikoff/android-components/blob/master/components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/matcher/UrlMatcher.kt),
```
Comment (psymoon) [inline docs/contribute/updating_tracking_protection_lists.md]: nit: sentence is too long, maybe something like:

It is a feature that prevents trackers from collecting your personal information like your browsing habits and interests. **`GeckoEngine` comes [built-in in Gecko(View)](https://mozilla.github.io/geckoview/javadoc/mozilla-central/org/mozilla/geckoview/ContentBlockingController.ContentBlockingException.html). `SystemEngine` we have [created our own implementation](https://github.com/vladikoff/android-components/blob/master/components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/matcher/UrlMatcher.kt),** [...]

File: docs/contribute/updating_tracking_protection_lists.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+which relays on different lists that indicates what should be considered as a tracker or not.
+
+## The lists
+
+These lists are kept outside of the Android Components repository in the [shavar-prod-lists](https://github.com/mozilla-services/shavar-prod-lists) repository.
```
Comment (psymoon) [inline docs/contribute/updating_tracking_protection_lists.md]: nit: The following lists [...] ?

File: docs/contribute/updating_tracking_protection_lists.md (hunk 1 above)
Comment (Amejia481) [reply docs/contribute/updating_tracking_protection_lists.md]: 👍

File: docs/contribute/updating_tracking_protection_lists.md (hunk 2 above)
Comment (Amejia481) [reply docs/contribute/updating_tracking_protection_lists.md]: 👍

**Comment by Amejia481** [comment]
@psymoon thanks for the review :)

**Comment by Amejia481** [comment]
bors r=@psymoon

**Pull request merged**